docs/src/developers_guide/testing_tools.rst

```diff
@@ -0,0 +1,70 @@
+.. include:: ../common_links.inc
+
+.. _testing_tools:
+
+Testing tools
+*************
+
+Iris has various internal convenience functions and utilities available to
+support writing tests. Using these makes tests quicker and easier to write, and
+also consistent with the rest of Iris (which makes it easier to work with the
+code). Most of these conveniences are accessed through the
+:class:`iris.tests.IrisTest` class, from which Iris' test classes inherit.
+
+Custom assertions
+=================
+
+:class:`iris.tests.IrisTest` supports a variety of custom unittest-style
+assertions, such as :meth:`~iris.tests.IrisTest_nometa.assertStringEqual`,
+:meth:`~iris.tests.IrisTest_nometa.assertArrayEqual`,
```

Comment:
It's a nice idea, but none of these links will work, because we don't let Sphinx include the test routines in the built API docs -- I think that is fixed [here](https://github.com/SciTools/iris/blob/v3.2.0/docs/src/sphinxext/generate_package_rst.py#L284-L285) ?

So, you could maybe make all these links target the codebase directly instead?
E.G. 
`such as :meth:'~iris.tests.IrisTest_nometa.assertStringEqual'`
--> `such as 'assertStringEqual <https://github.com/SciTools/iris/blob/main/lib/iris/tests/__init__.py#L264>'_`

Unfortunately though, those line-numbers won't be stable against code changes.
Though I guess you could invent a sphinx plugin for that ?...

Reply:
Does the `literalinclude` directive help?
https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-literalinclude

Reply:
Does it matter if we have links from here, or can we just double tick quote it and assume that anyone looking at the testing tools is happy to find them themselves in the codebase?

I'm not sure `literalinclude` does help @rcomer, unless I've misunderstood what you mean to do with it?

Reply:
Aside: I believe `assertStringEqual` could be replaced with [assertMultiLineEqual](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertMultiLineEqual).  In fact I made a [branch](https://github.com/SciTools/iris/compare/main...rcomer:remove-assertwarnsregexp) that does that.

Reply:
As discussed in Peloton, have laid out the breadcrumbs for devs to follow in the absence of handy links.

Reply:
@pp-mo are you happy with this as it stands now to go in?